### PR TITLE
Make all TextElement properties affect TextPresenter's render

### PR DESCRIPTION
**Pull request by SKProCH**
## What does the pull request do?
This PR makes the TextElement properties used in TextPresenter affect the render.  

## What is the current behavior?
Currently, if we have a `SolidColorBrush` in resources and updating his value TextBox will not be re-rendered with updated color  (However, the color has changed). After that i need to force TextBox to redraw yourself to view new color. 
Here is the Material.Avalonia theming issue - https://github.com/AvaloniaCommunity/Material.Avalonia/issues/200
Here is minimum reproduceable example - https://github.com/SKProCH/AvaloniaChangeColorRepro
(Background will display current color, but TextBox's text will not affected until i manually update it (by focusing him):
a542c969-b42d-4fc2-83ab-b75ef7017683

## What is the updated/expected behavior with this PR?
When i update Color property on SolidColorBrush binded to TextBox's Foreground - TextBox's color actually updated and TextBox re-rendered

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/AvaloniaUI/Avalonia?pullRequest=9631) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/AvaloniaUI/Avalonia?pullRequest=9631) before we can accept your contribution.<br/><br/>:x: SKProCH [sign now](https://cla.dotnetfoundation.org/AvaloniaUI/Avalonia?pullRequest=9631)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/AvaloniaUI/Avalonia?pullRequest=9631) it.</sub>

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0027088-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-nightly/nuget/v3/index.json) [PRBUILDID]

**Comment by robloo** [comment]
Good job tracking this one down!

File: src/Avalonia.Controls/Presenters/TextPresenter.cs [hunk 1]
```diff
@@ -110,7 +110,8 @@ public class TextPresenter : Control
 
         static TextPresenter()
         {
-            AffectsRender<TextPresenter>(CaretBrushProperty, SelectionBrushProperty);
+            AffectsRender<TextPresenter>(CaretBrushProperty, SelectionBrushProperty, 
+                TextElement.FontFamilyProperty, TextElement.FontSizeProperty, TextElement.FontStyleProperty, TextElement.FontWeightProperty, TextElement.FontStretchProperty, TextElement.ForegroundProperty);
```
Comment (robloo) [inline src/Avalonia.Controls/Presenters/TextPresenter.cs:114]: This is kind-of a nit, but if parameters are so long they span multiple lines there should be one on each line. This really helps with readability in this case.

```csharp
AffectsRender<TextPresenter>(
    CaretBrushProperty,
    SelectionBrushProperty, 
    TextElement.FontFamilyProperty,
    TextElement.FontSizeProperty,
    TextElement.FontStyleProperty,
    TextElement.FontWeightProperty,
    TextElement.FontStretchProperty,
    TextElement.ForegroundProperty);
```

Comment (SKProCH) [reply src/Avalonia.Controls/Presenters/TextPresenter.cs]: Yep, fixed that

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0027127-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-nightly/nuget/v3/index.json) [PRBUILDID]

**Comment by Gillibald** [comment]
Only brush properties need to be registered with AffectsRender. The underlying issue seems to be that the brushes color changes and not the brush itself. Therefore there is never a change notification and the TextPresenter isn't invalidated. AffectsRender subscribes to changes of the mutating brush and therefore the TextPresenter is invalidated.

**Comment by SKProCH** [comment]
@Gillibald done

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0027203-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-nightly/nuget/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0027209-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-nightly/nuget/v3/index.json) [PRBUILDID]

**Comment by grokys** [comment]
LGTM though this isn't my area. @Gillibald are we ok to merge? Looks like a pretty uncontroversial change.

**Review by Gillibald — APPROVED** [review]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0027628-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Pull request merged**